Compatibility let syntax (#1899)
ocaml/dune: doc/dune-files.rst

- ``(staged_pps <ppx-rewriters-and-flags>)`` is similar to ``(pps ...)``
  but behave slightly differently and is needed for certain ppx rewriters
  (see below for details)

> - ``compat`` is a special value that brings some of the newer OCaml

Reviewer: How about `compat-syntax`? or `future-syntax`? (that's similar to what python uses)

Author: I like `future-syntax`, let's go for that

Author: I'm using `future_syntax` to be consistent with the other cases

>   OCaml syntaxes to older compiler. See :ref:`Future syntax

Reviewer: ```suggestion
  OCaml syntaxes to older compilers. See :ref:`Future syntax
```

                 (((action (run ./pp.sh X=1 %{input-file})) foo bar))
                 (((action (run ./pp.sh X=2 %{input-file})) baz))))

.. _compat-pp:

Compat preprocessor
~~~~~~~~~~~~~~~~~~~

The ``compat`` preprocessing specification is equivalent to

> ``no_preprocessing`` when using a recent versions of the

Reviewer: ```suggestion
``no_preprocessing`` when using a recent version of the
```

(should this be "the latest"?)

Author: note that it can be several versions of the compiler

Author: I change this sentence

compiler. When using an older one, it is a shim preprocessor that

> backports some of the newer syntaxes. This allows you to use some of

Reviewer: ```suggestion
backports some of the newer syntax elements. This allows you to use some of
```
